pages/api/roles/index.ts

```diff
@@ -68,38 +69,16 @@ async function handler(req: NextApiRequest, res: NextApiResponse) {
 }
 
 async function handleGet(req: NextApiRequest, res: NextApiResponse<ApiResponse<RoleListData[]>>) {
-  const { companyId, searchQuery } = req.query as RoleQueryParams;
-
-  const queryCompanyId = typeof companyId == 'string' && canBecomeInteger(companyId) ? Number(companyId) : undefined;
-
-  const searchWords = typeof searchQuery == 'string' && !isEmpty(searchQuery) ? searchQuery.trim().split(/\s+/) : [];
-
-  const searchInts = searchWords
-    .filter(canBecomeInteger)
-    .map(Number)
-    .filter((year) => year >= MIN_ROLE_YEAR);
-  const queryYear = searchInts.length > 0 ? searchInts[0] : undefined;
-
-  const queryWords = searchWords
-    .filter((word) => !(canBecomeInteger(word) && Number(word) >= MIN_ROLE_YEAR))
-    .map((word) => ({
-      title: {
-        contains: word,
-        mode: Prisma.QueryMode.insensitive,
-      },
-    }));
-
-  if (!queryCompanyId) {
-    res.status(HttpStatus.OK).json(createJsonResponse([]));
-    return;
-  }
+  const queryParams: RoleQueryParams = parseGetQueryParams(req);
+  const roleTitleFilters = makeRoleTitleFilters(queryParams.searchWords);
+  const roleYearFilters = makeRoleYearFilters(queryParams.searchWords);
 
   const roles: RoleListData[] = await prisma.role.findMany({
     where: {
       isVerified: true,
-      companyId: queryCompanyId,
-      AND: queryYear ? [{ year: { equals: queryYear } }] : undefined,
-      OR: queryWords.length > 0 ? queryWords : undefined,
+      companyId: queryParams.companyId,
+      AND: roleYearFilters,
```

Comment:
```suggestion
      OR: roleYearFilters,
```

Would a OR search make more sense here? If i type '2023 2024', i will get no results according to this logic.